src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/CodeBasedDependencyAlgorithm.cs

```diff
@@ -23,6 +23,12 @@ public static void AddDependenciesDueToMethodCodePresence(ref DependencyList dep
 
             factory.InteropStubManager.AddDependeciesDueToPInvoke(ref dependencies, factory, method);
 
+            if (method.OwningType is MetadataType mdOwningType)
+            {
+                dependencies ??= new DependencyList();
+                dependencies.Add(factory.ModuleUse(mdOwningType.Module), "Module with a used method");
```

Comment:
This will add a lot of dependencies that are useless in nearly 100% cases. These dependencies only matter when the module has static constructor. Should we skip adding these dependencies when the module does not have static constructor (e.g. return null from `ModuleUse`)?

Reply:
Yeah, maybe we can just get rid of ModuleUse and emit a dependency on the cctor directly.